samples: net: cloud: tagoio_http_post: Undefined initialization levels used.

**Describe the bug**
There are some hardware combinations (board/shields) that report at link phase:
```
zephyr-sdk-0.12.0.b/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.2.0/../../../../arm-zephyr-eabi/bin/ld: Undefined initialization levels used.
Memory region         Used Size  Region Size  %age Used
           FLASH:      123192 B         1 MB     11.75%
            SRAM:       50319 B       256 KB     19.20%
        IDT_LIST:         104 B         2 KB      5.08%
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Try build with:

`west build -b nrf52840dk_nrf52840 samples/net/cloud/tagoio_http_post -- -DSHIELD=esp_8266_arduino -DOVERLAY_CONFIG=overlay-wifi.conf`

**Impact**
Annoyance

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.12.0-beta-1
 - Commit SHA f004410aa1

--- a/samples/net/cloud/tagoio_http_post/README.rst
+++ b/samples/net/cloud/tagoio_http_post/README.rst
@@ -1,7 +1,7 @@
 .. _cloud-tagoio-http-post-sample:
 
-Socket TagoIO HTTP Client
-#########################
+TagoIO IoT Cloud HTTP Sample
+############################
 
 Overview
 ********

--- a/samples/net/cloud/tagoio_http_post/prj.conf
+++ b/samples/net/cloud/tagoio_http_post/prj.conf
@@ -42,3 +42,6 @@ CONFIG_NET_HTTP_LOG_LEVEL_DBG=n
 # TagoIO API
 CONFIG_TAGOIO_DEVICE_TOKEN="<your device token API>"
 CONFIG_CBPRINTF_FP_SUPPORT=y
+
+# Fix #30029 - Undefined initialization levels used
+CONFIG_PINMUX=y

